instrumentation/rocketmq-client-4.8/testing/src/main/groovy/io/opentelemetry/instrumentation/rocketmq/AbstractRocketMqClientTest.groovy

```diff
@@ -66,17 +69,26 @@ abstract class AbstractRocketMqClientTest extends InstrumentationSpecification {
     BaseConf.deleteTempDir()
   }
 
+  def setup() {
+    tracingMessageListener.reset()
+  }
+
   def "test rocketmq produce callback"() {
+    CompletableFuture<SendResult> result = new CompletableFuture<>()
     when:
     producer.send(msg, new SendCallback() {
       @Override
       void onSuccess(SendResult sendResult) {
+        result.complete(sendResult)
       }
 
       @Override
       void onException(Throwable throwable) {
+        result.completeExceptionally(throwable)
       }
     })
+    result.get(10, TimeUnit.SECONDS).sendStatus == SendStatus.SEND_OK
+    tracingMessageListener.waitForMessages()
```

Comment:
otherwise someone (me) might think they were cleaning up old/unnecessary code and remove it 😄 
```suggestion
    // waiting longer than assertTraces below does on its own because of CI flakiness
    tracingMessageListener.waitForMessages()
```